vhost-device-scmi/src/devices/iio.rs

```diff
@@ -280,13 +398,110 @@ impl SensorT for IIOSensor {
         let mut result = vec![];
         for axis in &self.axes {
             let value = self.read_axis(axis)?;
-            result.push(MessageValue::Unsigned((value & 0xFFFFFFFF) as u32));
-            result.push(MessageValue::Unsigned((value >> 32) as u32));
+            result.push(MessageValue::Signed((value & 0xFFFFFFFF) as i32));
+            result.push(MessageValue::Signed((value >> 32) as i32));
```

Comment:
Are these changes related?
Or it's a fix of a pre-existing issue?

Reply:
Yes, it is a fix of a pre-existing issue which I have discussed with [mz-pdm](https://github.com/mz-pdm).
For more details, you can look this [conversation](https://github.com/rust-vmm/vhost-device/pull/710#discussion_r1722857750)

Reply:
So, please put in a separate commit explaining what you're fixing. (same PR is fine)

Reply:
In latest push, I have separated it.